metadata-ingestion/tests/unit/test_mssql.py

```diff
@@ -299,3 +299,44 @@ def test_stored_procedure_vs_direct_query_compatibility(mssql_source):
     # Verify database_name is properly set
     assert sp_step["database_name"] == "test_db"
     assert direct_step["database_name"] == "test_db"
+
+
+def test_get_sql_alchemy_url_ignores_config_db_when_override_is_provided():
+    """Test that the database name is ignored when an override is provided"""
+    override_db = "override_db"
+    config_defined_db = "config_defined_connection_db"
+
+    config = SQLServerConfig(
+        host_port="localhost:1433",
+        username="test",
+        password="test",
+        use_odbc=True,
+        uri_args={"database": config_defined_db, "driver": "some_driver_value"},
+    )
+
+    # Mock to avoid DB connections
+    with (
+        patch("datahub.ingestion.source.sql.sql_common.SQLAlchemySource.__init__"),
+        patch(
+            "datahub.ingestion.source.sql.mssql.source.SQLServerSource._database_names_from_engine"
+        ) as mock_db_names,
+        patch(
+            "datahub.ingestion.source.sql.mssql.source.SQLServerSource._inspector_for_database"
+        ),
+    ):
+        mock_db_names.return_value = [override_db]
+        source = SQLServerSource(config, MagicMock())
+
+        # Call the method with the override
```

Comment:
```suggestion
        # Call the method without the override
```

we override when `current_db` is passed, and we don't override if `current_db` is missed, right?